src/Core/Framework/Webhook/Service/WebhookCleanup.php

```diff
@@ -36,22 +38,31 @@ public function removeOldLogs(): void
             return;
         }
 
+        // Delete older webhook log entries where the webhook won't be called anymore
+        $this->deleteLogsOlderThanWithStatus($entryLifetimeSeconds, WebhookEventLogDefinition::STATUS_SUCCESS, WebhookEventLogDefinition::STATUS_FAILED);
+        // after double the entry lifetime, we also delete queued entries,
+        // because we assume they are stuck in queued state (as we rely on message retry to retry failed webhooks)
+        $this->deleteLogsOlderThanWithStatus($entryLifetimeSeconds * 2, WebhookEventLogDefinition::STATUS_QUEUED);
```

Comment:
I'm wondering if maybe we should have a separate task that looks for webhooks stuck in a queued state and sets them to `STATUS_FAILED`. That would maybe would allow a bit more customisation, eg lifetime or destination status, could dispatch an event which might allow other to retry on certain conditions etc. Or we can then log that some tasks got lost. This just feels a bit too magic.

Is that overkill ?

Reply:
But just to be clear here: this pr is now fixing two separate problems right?

1. Worker dies and event log stays in queued status and never gets cleaned up
2. Event log is somehow removed and prevents webhook from being retried

Under what scenario is number 2 happening? Someone manually deleting logs from the table? Or when lifetime for removing is really low?